### PR TITLE
WIP: OpenTelemetry build tags for 5.2

### PR DESCRIPTION
**Pull request by titpetric**
Add `v52` build tags over otel/graphql extensions. This allows us to have an acceptance test for code on master that can be built with the 5.0 LTS release toolchains and go.mod/sum.

This PR: https://github.com/TykTechnologies/exp/actions/runs/5801476643/job/15725906504
Master: https://github.com/TykTechnologies/exp/actions/runs/5786664992

Master:

```
Error: apidef/adapter/openapi.go:15:2: no required module provides package github.com/TykTechnologies/graphql-go-tools/pkg/openapi; to add it:
	go get github.com/TykTechnologies/graphql-go-tools/pkg/openapi
Error: apidef/oas/old_oas.go:6:2: no required module provides package github.com/TykTechnologies/kin-openapi/openapi3; to add it:
	go get github.com/TykTechnologies/kin-openapi/openapi3
Error: internal/otel/otel.go:[8](https://github.com/TykTechnologies/exp/actions/runs/5786664992/job/15681892094#step:6:9):2: no required module provides package github.com/TykTechnologies/opentelemetry/config; to add it:
	go get github.com/TykTechnologies/opentelemetry/config
Error: internal/otel/otel.go:[9](https://github.com/TykTechnologies/exp/actions/runs/5786664992/job/15681892094#step:6:10):2: no required module provides package github.com/TykTechnologies/opentelemetry/semconv/v1.0.0; to add it:
	go get github.com/TykTechnologies/opentelemetry/semconv/v1.0.0
Error: internal/otel/otel.go:[10](https://github.com/TykTechnologies/exp/actions/runs/5786664992/job/15681892094#step:6:11):2: no required module provides package github.com/TykTechnologies/opentelemetry/trace; to add it:
	go get github.com/TykTechnologies/opentelemetry/trace
Error: Process completed with exit code 1.
```

Remaining on this PR:

```
Error: apidef/adapter/openapi.go:15:2: no required module provides package github.com/TykTechnologies/graphql-go-tools/pkg/openapi; to add it:
	go get github.com/TykTechnologies/graphql-go-tools/pkg/openapi
Error: apidef/oas/old_oas.go:6:2: no required module provides package github.com/TykTechnologies/kin-openapi/openapi3; to add it:
	go get github.com/TykTechnologies/kin-openapi/openapi3
Error: Process completed with exit code 1.
```

WIP POC

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/5397/merge
Commit:  
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5801279221)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5397/merge
Commit: b4352db7c216a6a0148cf37b7d270017c552cfdc 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5801368132)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5397/merge
Commit: e758d74c4c8ae52ebda2c86b2a32b1e198ad118b 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5801448302)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5397/merge
Commit: d21e45a6c398aaf46a54ab6428e85f3cee018ad1 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5801501932)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5397/merge
Commit: 08798c27e00a90870333ff112677388fc47aea51 
Triggered by: pull_request (@titpetric)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5806102455)

File: gateway/mw_graphql.go [hunk 1]
```diff
@@ -286,44 +277,6 @@ func (m *GraphQLMiddleware) validateRequest(w http.ResponseWriter, gqlRequest *g
 	return nil, http.StatusOK
 }
 
-func (m *GraphQLMiddleware) validateRequestWithOtel(ctx context.Context, w http.ResponseWriter, req *gql.Request) (error, int) {
-	m.Spec.GraphQLExecutor.OtelExecutor.SetContext(ctx)
```
Comment (titpetric) [inline gateway/mw_graphql.go:290]: @tbuchaillot @sredxny this is a code smell btw, what if there are multiple requests in flights, as they are parallel (goroutine per request), the context state is not going to be correct. Individual function invocations below need to take ctx and not hold it in shared state. https://github.com/sivchari/containedctx

Comment (tbuchaillot) [reply gateway/mw_graphql.go:290]: cc @kofoworola @pvormste

**Pull request closed without merging**